modules/platform/nuxeo-platform-usermanager/src/main/java/org/nuxeo/ecm/platform/usermanager/UserManagerImpl.java

```diff
@@ -1259,7 +1259,11 @@ protected DocumentModel getUserModel(String userName, DocumentModel context, boo
         }
 
         try (Session userDir = dirService.open(userDirectoryName, context)) {
-            return userDir.getEntry(userName, fetchReferences);
+            DocumentModel userModel = userDir.getEntry(userName, fetchReferences);
+            if(userModel!=null) {
+                userModel.putContextData("userHasPartialContent", !fetchReferences);
```

Comment:
Please put `"userHasPartialContent"` in a constant in `UserManagerImpl` (so, internal for now), and use it also from `NuxeoPrincipalJsonWriter`.